Superposition/ReferenceImplementation.qs

```diff
@@ -429,5 +429,19 @@ namespace Quantum.Kata.Superposition {
             }
         }
     }
-    
+
+    // ------------------------------------------------------
+    // Task 16**. Hardy State
```

Comment:
Please also add this task to the Jupyter Notebook version of this kata, so that all solvers can enjoy it :-)